.github/workflows/cd.yml

```diff
@@ -17,12 +22,15 @@ on:
 jobs:
   push_changes:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inpupts.data_project_dir }}
     name: Push changes to Workspace
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          fetch-depth: 0
+          fetch-depth: 300
```

Comment:
This number is a bit arbitrary. Maybe we could allow override full (0) with a parameter

Reply:
Let's decide it later.

BTW this 300 means last 300 commits or what?